entrypoint/entrypoint/orchestrator.py

```diff
@@ -221,17 +221,128 @@ def invoke_sbomgen(args) -> int:
     return ret
 
 
-def invoke_inspector_scan(src_sbom, dst_scan):
-    aws_cli_args = ["inspector-scan", "scan-sbom",
-                    "--sbom", f"file://{src_sbom}",
-                    "--output-format", "CYCLONE_DX_1_5",
-                    ">", f"{dst_scan}"
-                    ]
+def invoke_inspector_scan(src_sbom, dst_scan, max_api_calls):
+    MAX_COMPONENTS_PER_SCAN = 5000
+
+    with open(src_sbom, 'r', encoding='utf-8') as f:
+        sbom = json.load(f)
+
+    sbom_components = sbom['components'].copy()
+    num_parts = math.ceil(len(sbom_components) / MAX_COMPONENTS_PER_SCAN)
+    require_true(
+        num_parts <= max_api_calls,
+        "SBOM components number exceed scan limit"
+    )
+
+    if num_parts <= 1:
+        aws_cli_args = ["inspector-scan", "scan-sbom",
+            "--sbom", f"file://{src_sbom}",
+            "--output-format", "CYCLONE_DX_1_5",
+            ">", f"{dst_scan}"
+        ]
+
+        ret = executor.invoke_command("aws", aws_cli_args)
 
-    ret = executor.invoke_command("aws", aws_cli_args)
-    if ret != 0:
         return ret
 
+    # Runs multiple API calls and combine the results
+    summery = {}
```

Comment:
This looks like the start of the "chunking" path.
Consider separating chunking logic into its own functions.